### DISCUSSION TITLE
Testing File Uploads in Deep Arrays

### DISCUSSION THREAD

**Discussion by mrfelipemartins** [livewire/livewire#7979] [7. Bugs (archived)]
### Livewire version

v3.4.5

### Laravel version

v10.44.0

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Create a Livewire component that accepts file uploads, specifically designed to handle files provided in a nested array structure (e.g., for handling multiple file uploads within grouped form data).
2. Write a test for the Livewire component, attempting to simulate the file upload process. The test should utilize Livewire's testing utilities to set the files in a nested array format, as shown below:
```php
Livewire::test(UploadPage::class)
    ->set('form.files', [
        [
            'name' => 'file1.txt',
            'file' => UploadedFile::fake()->create('file1.txt'),
        ],
        [
            'name' => 'file2.txt',
            'file' => UploadedFile::fake()->create('file2.txt'),
        ],
    ])
    ->call('upload')
    ->assertHasNoErrors();
```

3. Run the test.

**Expected Result:**

The test should complete successfully, with Livewire correctly handling the file uploads in a deep array, allowing for assertions to be made about the upload process without any errors.

**Actual Result:**

The test throws an error when attempting to handle file uploads provided in a deep array structure.

```
ErrorException: Trying to access array offset on value of type null
```

### Repository or Wirebox URL

https://github.com/mrfelipemartins/livewire-tests

### Additional information

The issue appears to be specific to handling file uploads within deep arrays during the testing process. The documentation does not explicitly cover this scenario, leading to potential misunderstandings or misconfigurations.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by kerwin-wildsea** [comment]
Just found this report after a couple hours of debugging. 
I wasn't able to find out why the snapshot entry isn't populated. It indeed only happens in testing, which makes it even harder to find (and weirder). 

Inside the laravel source code, running `request('components')` results in the following for me:
```
array:1 [
  0 => array:3 [
    "snapshot" => "null"
    "calls" => array:1 [
      0 => array:3 [
        "method" => "submit"
        "params" => []
        "path" => ""
      ]
    ]
    "updates" => []
  ]
] // vendor/livewire/livewire/src/Mechanisms/HandleRequests/HandleRequests.php:93
```

PS: Main reason for comment is to bump the thread 🙂

**Comment by Lintume** [comment]
Same problem here, spent on resolving several days... nothing. Found workaround, if someone interested I can share

**Comment by ao-jhelmich** [comment]
bumping this

getting the same thing when trying to set a array with data in my test.


My working working around for now is instead of doing

```php
->set('state', $data)
```

I do

```php
->set(Arr::prependKeysWith($data, 'state.'))
```
